server/tests/materials.test.ts

```diff
@@ -0,0 +1,640 @@
+import { IPayload } from '../utils/jwtUtils'
+import jwt from 'jsonwebtoken'
+
+import { Material } from '../db/models/material'
+
+dotenv.config()
+const app = require('../app')
+
+jest.mock('@google-cloud/storage', () => {
+    const createWriteStreamMock = jest.fn()
+    createWriteStreamMock.mockReturnValue({
+        on: jest.fn((event: string, callback: Function) => {
+            if (event === 'error') {
+                return
+            } else if (event === 'finish') {
+                return callback()
+            }
+        }),
+        end: jest.fn(),
+    })
+
+    return {
+        Storage: jest.fn(() => ({
+            bucket: jest.fn(() => ({
+                file: jest.fn(() => ({
+                    createWriteStream: createWriteStreamMock,
+                    delete: jest.fn(),
+                })),
+            })),
+        })),
+    }
+})
+
+const materials = [
+    new Material({
+        _id: '615e2f3b1d9f9b2b4c9e9b1a',
+        __v: 0,
+        title: 'Libro 1',
+        description: 'Descripción 1',
+        author: 'maria',
+        price: 12,
+        currency: 'EUR',
+        purchasers: ['pepe'],
+        type: 'book',
+        file: 'https://mockedFile.txt',
+    }),
+    new Material({
+        _id: '615e2f3b1d9f9b2b4c9e9b1b',
+        __v: 0,
+        title: 'Libro 2',
+        description: 'Descripción 2',
+        author: 'pepe',
+        price: 15,
+        currency: 'EUR',
+        purchasers: ['maria'],
+        type: 'book',
+        file: 'https://file2.json',
+    }),
+    new Material({
+        _id: '615e2f3b1d9f9b2b4c9e9b1c',
+        __v: 0,
+        title: 'Libro 3',
+        description: 'Descripción 3',
+        author: 'marta',
+        price: 32,
+        currency: 'USD',
+        purchasers: [],
+        type: 'book',
+        file: 'https://file3.json',
+    }),
+]
+
+const URL_BASE = '/api/v1/materials'
+const JSON_WEB_TOKEN =
+    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjp7Il9pZCI6IjY1NzFiNzNjMjUyYWRlZWI4MDczODNjNiIsImZpcnN0TmFtZSI6Ik5vbWJyZSIsImxhc3ROYW1lIjoiQXBlbGxpZG8iLCJ1c2VybmFtZSI6Im1hcmlhIiwicGFzc3dvcmQiOiJjb250cmFzZW5hMTIzIiwiZW1haWwiOiJ1c3VhcmlvQGV4YW1wbGUuY29tIiwicGxhbiI6IlBSRU1JVU0iLCJyb2xlIjoiVVNFUiJ9LCJpYXQiOjE3MDIwNjI5MzksImV4cCI6MTczMzU5ODkzOX0.Hu0f9BoIzULvkZzfCWGvSSxofUTABK6D4PeGuNw_438'
+const UNAUTHORIZED_JWT =
+    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjp7Il9pZCI6IjY1NzFiNzNjMjUyYWRlZWI4MDczODNjNiIsImZpcnN0TmFtZSI6Ik5vbWJyZSIsImxhc3ROYW1lIjoiQXBlbGxpZG8iLCJ1c2VybmFtZSI6Im1hcnRhIiwicGFzc3dvcmQiOiJjb250cmFzZW5hMTIzIiwiZW1haWwiOiJ1c3VhcmlvQGV4YW1wbGUuY29tIiwicGxhbiI6IlBSRU1JVU0iLCJyb2xlIjoiVVNFUiJ9LCJpYXQiOjE3MDIwNjI5MDAsImV4cCI6MTczMzU5ODkwMH0.Vvw5IZy7u35VBuodZTauln1Nf7PDDaOcNQbHuIE4F5c'
+const JWT_SECRET = 'secret'
+
+describe('Materials API', () => {
+    describe('GET /materials/:me', () => {
+        let findMaterialsByUsernameMock: jest.SpyInstance
+        let payload: IPayload = jwt.verify(JSON_WEB_TOKEN, JWT_SECRET, {
```

Comment:
I would separate the verification of token from the unit test, althought this only exists here. This separation lets accomplish the principle of single responsability

Reply:
You are absolutely right. I will separate the verification logic from the test token.